tool/checkstyle/rules.bzl

```diff
@@ -66,34 +62,39 @@ def _checkstyle_test_impl(ctx):
         [file.path for file in files]
     )
 
+    checkstyle_wrapper = ctx.actions.declare_file("%s.kt" % ctx.attr.name)
     ctx.actions.expand_template(
-        template = ctx.file._checkstyle_py_template,
-        output = ctx.outputs.checkstyle_script,
+        template = ctx.file._checkstyle_kt_template,
+        output = checkstyle_wrapper,
         substitutions = {
             "{command}" : cmd,
-            "{allow_failure}": str(int(ctx.attr.allow_failure)),
         },
         is_executable = True,
     )
 
-    files = [ctx.outputs.checkstyle_script] + ctx.files._license_files + files + ctx.files._classpath + inputs
+    files = [checkstyle_wrapper] + ctx.files._license_files + files + ctx.files._classpath + inputs
     runfiles = ctx.runfiles(
         files = files,
         collect_data = True,
     )
     return DefaultInfo(
-        executable = ctx.outputs.checkstyle_script,
-        files = depset(files),
+        executable = checkstyle_wrapper,
         runfiles = runfiles,
     )
 
-checkstyle_test = rule(
-    implementation = _checkstyle_test_impl,
-    test = True,
+_checkstyle_wrapper = rule(
+    implementation = _checkstyle_wrapper_impl,
```

Comment:
This rule can no longer be run as a test—it needs to be compiled as a `kt_jvm_test`—, so it no longer gets the suffix.